### PR TITLE
[quant] Add reference quantized linear/linear_relu module

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48192 [quant] Add reference quantized linear/linear_relu module**

Summary:
Some backend don't have packed weight for linear/linear_relua, this adds a reference
module that does not contain packed weight for linear/linear_relu

Test Plan:
test_quantized_module.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25061645](https://our.internmc.facebook.com/intern/diff/D25061645)

### CODE REVIEW

File: torch/nn/intrinsic/quantized/modules/__init__.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
-
-from .linear_relu import LinearReLU
+from .linear_relu import LinearReLU, LinearReLUBackendIndependent
 from .conv_relu import ConvReLU1d, ConvReLU2d, ConvReLU3d
 from .bn_relu import BNReLU2d, BNReLU3d
 
 __all__ = [
     'LinearReLU',
+    'LinearReLUBackendIndependent',
```
Comment (raghuramank100) [inline torch/nn/intrinsic/quantized/modules/__init__.py:7]: NIT: Should we create a namespace nn.quantized.reference and have Linear, LinearRelu in there? The name BackendIndependent could be confusing to external users. Also, if we dont intend this for direct use, should we prefix with an _?, i.e have:
nn.quantized._reference.LinearRelu

File: torch/nn/quantized/modules/linear.py [hunk 2]
```diff
@@ -123,11 +124,12 @@ class Linear(torch.nn.Module):
         >>> print(output.size())
         torch.Size([128, 30])
     """
-    _version = 3
+    _version = 4
```
Comment (raghuramank100) [inline torch/nn/quantized/modules/linear.py:127]: Why is the version number bumped up here?

File: torch/nn/intrinsic/quantized/modules/__init__.py (hunk 1 above)
Comment (jerryzh168) [reply torch/nn/intrinsic/quantized/modules/__init__.py:7]: Sure

File: torch/nn/quantized/modules/linear.py (hunk 2 above)
Comment (jerryzh168) [reply torch/nn/quantized/modules/linear.py:127]: sorry, this is left over from previous changes, will revert

File: torch/nn/intrinsic/quantized/modules/__init__.py (hunk 1 above)
Comment (jerryzh168) [reply torch/nn/intrinsic/quantized/modules/__init__.py:7]: @raghuramank100 do you also suggest to change backend_independent to reference in the api?

Comment (jerryzh168) [reply torch/nn/intrinsic/quantized/modules/__init__.py]: changed to reference

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**